### PR TITLE
final few deprecations removal in a batch

### PR DESCRIPTION
**Pull request by ndelangen**
removal of the last few deprecations for 7.0

### CODE REVIEW

**Comment by ndelangen** [comment]
@shilman Do you think you can help me understand why https://app.circleci.com/pipelines/github/storybookjs/storybook/30614/workflows/4a03924c-4716-4d9e-ac10-c499ba5fdae8/jobs/429471 is failing? Seems the background color isn't applied anymore, but I can't seem to find the reason why.

**Comment by ndelangen** [comment]
@tmeasday you marked adding unknown globals as deprecated, but it looks like removing that actually breaks addon-backgrounds?

File: code/lib/store/src/GlobalsStore.test.ts [hunk 1]
```diff
@@ -48,14 +48,13 @@ describe('GlobalsStore', () => {
 
       store.update({ baz: 'bing' });
       expect(store.get()).toEqual({ foo: 'bar', baz: 'bing' });
-
-      // NOTE: this is currently allowed but deprecated.
-      store.update({ random: 'value' });
-      expect(store.get()).toEqual({ foo: 'bar', baz: 'bing', random: 'value' });
```
Comment (ndelangen) [inline code/lib/store/src/GlobalsStore.test.ts:54]: This is where the global update code use to be @shilman

**Comment by tmeasday** [comment]
> @tmeasday you marked adding unknown globals as deprecated, but it looks like removing that actually breaks addon-backgrounds?

Well yeah, I guess backgrounds should export a `globalTypes` definition for `BACKGROUNDS_PARAM_KEY` (`'backgrounds'`)

**Comment by ndelangen** [comment]
That fixed it @tmeasday ♥️

**Comment by ndelangen** [comment]
@shilman mentioned that he wanted to discuss this deprecation further.

He said that silently ignoring new incoming properties was not a good idea. I tend to agree.
We should either fail hard or continue to warn.

**Comment by ndelangen** [comment]
I'll add a bit of code that throws an error when a property is filtered out

**Comment by shilman** [comment]
@tmeasday can you chime in on why we want to force addons to declare their `globalTypes` before using them? it makes sense intuitively, but is there some specific use case or problem that you have in mind? Just curious to help me understand the change better. Thanks!

**Comment by tmeasday** [comment]
@shilman, [according to the original commit](https://github.com/storybookjs/storybook/commit/3aa4fc897ae26c0574b3263c52197ccd61e13059):

> This ensures weird stuff doesn't happen when we change arg/global names.

**Comment by shilman** [comment]
@tmeasday thanks, just checking. i think we should do it FWIW.

**Pull request merged**